### PR TITLE
Common nav

### PR DESCRIPTION
**Pull request by MukulAmbulgekar**
This PR includes following -

1. Implements mega-nav header/footer for the docs app.
2. Integrates Greg's refined mega-nav.xqy module.
3. Fixes broken search regression on dmc-staging app (changes in`src/config/template.xhtml` and `src/css/screen.css` )

Note that in `src/lib/mega-nav.xqy`, I have updated `default-user-role` and `authenticated-user-role` to `dmc-user`, as Dave confirmed in prod environment we use this role.  If this role is not present in you dev env you may get error, please create this role or update `mega-nav.xqy` to `admin`, if you are using admin to deploy the dev app.

### CODE REVIEW

**Comment by kcoleman-marklogic** [comment]
I'm a little concered about the default-user-role and authenticated-user-role being set to a role that is not created by default when you deploy RunDMC. Would it make sense to add that to deployment configuration, or would we rather not?

**Comment by dmcassel** [comment]
The current default when deploying via Roxy is the admin user. Yes, it would be better to update the config to create the same user & role, then use that. That would be done in deploy/ml-config.xml.

**Comment by MukulAmbulgekar** [comment]
What about permissions for this role in the ml-config? Should I inherit from [${app-name}-role](https://github.com/marklogic-community/RunDMC/blob/master/deploy/default.properties#L103) (I think this role is created by default when you deploy RunDMC app and it has insert/update permissions ) ? Or should I assign 'admin' role to the 'dmc-user' role ?

**Comment by MukulAmbulgekar** [comment]
Eduardo wants these changes to be deployed to the staging for review, I pushed updates to ml-config.xml to create 'dmc-user' while deploying the RunDMC. 

@dmcassel as we discussed earlier, 'dmc-user' has admin role on the prod instance, so I kept the same in the ml-config as well. Let me know if you think, I should make any updates to the 'dmc-user' role config. Thanks.

File: src/apidoc/config/template.xhtml [hunk 1]
```diff
@@ -1,159 +1,163 @@
...
-    <!-- START ORBIT CSS -->
-    <ml:common-nav-css/>
-    <!-- END ORBIT CSS -->
-    <link href="/css/v-38/apidoc-screen.css"
-          rel="stylesheet" type="text/css" media="screen" />
-    <link href="/css/v-33/apidoc.css"
-          rel="stylesheet" type="text/css" media="screen" />
-    <link href="/css/v-1/apidoc_print.css"
-          rel="stylesheet" type="text/css" media="print" />
+xmlns:ml="http://developer.marklogic.com/site/internal"
```
Comment (dmcassel) [inline src/apidoc/config/template.xhtml]: Looks like you lost a lot of the indentation here in the header. Could you restore that, please?

File: src/css/apidoc-screen.css [hunk 2]
```diff
@@ -572,10 +573,10 @@ nav ul {
   line-height: 100%;
   margin: 0 auto;
   position: relative;
-  width: 97%
+  /*width: 97%*/
```
Comment (dmcassel) [inline src/css/apidoc-screen.css]: looks like this is no longer being used; please delete

File: src/css/apidoc.css [hunk 3]
```diff
@@ -195,57 +197,57 @@ ol ol {
...
-}
-#page_content dd pre:first-child {
-    background-color: transparent;
-    padding: 0;
-}
-#api_sub {
-    color: #333333;
-    font-size: 0.9em;
-}
+    #page_content ul {
```
Comment (dmcassel) [inline src/css/apidoc.css]: Please make the indentation consistent with the rest of the code. Looks like nothing in this section changed except for indentation.

File: src/css/apidoc.css [hunk 4]
```diff
@@ -195,57 +197,57 @@ ol ol {
...
+        background: #fff;
+        height: 100%;
+        padding: 16px 0 16px 0;
+    }
 
 /* This needs some non-standard properties,
  * or else the user will accidentally select adjoining text.
  */
-#splitter {
+ #splitter {
```
Comment (dmcassel) [inline src/css/apidoc.css]: indentation

File: src/css/apidoc.css [hunk 5]
```diff
@@ -793,9 +795,9 @@ span.highlightWidget {
     #search-header form input[type="text"]:focus {
         width: 144px;
     }
-    #search_suggest_container {
+    /*#search_suggest_container {
```
Comment (dmcassel) [inline src/css/apidoc.css]: if we don't need this anymore, let's delete it

**Comment by dmcassel** [comment]
Found a few minor code-cleanliness things to fix, otherwise looks good.

**Pull request merged**